pkg/varlinkapi/pods.go

```diff
@@ -0,0 +1,237 @@
+		for ctr, reason := range ctrErrs {
+			ctrErr := iopodman.PodContainerErrorData{Containerid: ctr, Reason: reason.Error()}
+			containerErrs = append(containerErrs, ctrErr)
+		}
+		return call.ReplyPodContainerError(pod.ID(), containerErrs)
+	}
+
+	return call.ReplyStopPod(pod.ID())
+}
+
+// RestartPod ...
+func (i *LibpodAPI) RestartPod(call iopodman.VarlinkCall, name string) error {
+	pod, err := i.Runtime.LookupPod(name)
+	if err != nil {
+		return call.ReplyPodNotFound(name)
+	}
+	ctrErrs, err := pod.Restart(getContext())
+	if err != nil && ctrErrs == nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	if ctrErrs != nil {
+		containerErrs := make([]iopodman.PodContainerErrorData, len(ctrErrs))
+		for ctr, reason := range ctrErrs {
+			ctrErr := iopodman.PodContainerErrorData{Containerid: ctr, Reason: reason.Error()}
+			containerErrs = append(containerErrs, ctrErr)
+		}
+		return call.ReplyPodContainerError(pod.ID(), containerErrs)
+	}
+
+	return call.ReplyRestartPod(pod.ID())
+}
+
+// KillPod kills the running containers in a pod.  If you want to use the default SIGTERM signal,
+// just send a -1 for the signal arg.
+func (i *LibpodAPI) KillPod(call iopodman.VarlinkCall, name string, signal int64) error {
+	killSignal := uint(syscall.SIGTERM)
+	if signal != -1 {
+		killSignal = uint(signal)
+	}
+
+	pod, err := i.Runtime.LookupPod(name)
+	if err != nil {
+		return call.ReplyPodNotFound(name)
+	}
+	ctrErrs, err := pod.Kill(killSignal)
+	if err != nil && ctrErrs == nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	if ctrErrs != nil {
+		containerErrs := make([]iopodman.PodContainerErrorData, len(ctrErrs))
+		for ctr, reason := range ctrErrs {
+			ctrErr := iopodman.PodContainerErrorData{Containerid: ctr, Reason: reason.Error()}
+			containerErrs = append(containerErrs, ctrErr)
+		}
+		return call.ReplyPodContainerError(pod.ID(), containerErrs)
+	}
+
+	return call.ReplyKillPod(pod.ID())
+}
+
+// PausePod ...
+func (i *LibpodAPI) PausePod(call iopodman.VarlinkCall, name string) error {
+	pod, err := i.Runtime.LookupPod(name)
+	if err != nil {
+		return call.ReplyPodNotFound(name)
+	}
+	ctrErrs, err := pod.Pause()
+	if err != nil && ctrErrs == nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	if ctrErrs != nil {
+		containerErrs := make([]iopodman.PodContainerErrorData, len(ctrErrs))
+		for ctr, reason := range ctrErrs {
+			ctrErr := iopodman.PodContainerErrorData{Containerid: ctr, Reason: reason.Error()}
+			containerErrs = append(containerErrs, ctrErr)
+		}
+		return call.ReplyPodContainerError(pod.ID(), containerErrs)
+	}
+
+	return call.ReplyPausePod(pod.ID())
+}
+
+// UnpausePod ...
+func (i *LibpodAPI) UnpausePod(call iopodman.VarlinkCall, name string) error {
+	pod, err := i.Runtime.LookupPod(name)
+	if err != nil {
+		return call.ReplyPodNotFound(name)
+	}
+	ctrErrs, err := pod.Unpause()
+	if err != nil && ctrErrs == nil {
+		return call.ReplyErrorOccurred(err.Error())
+	}
+	if ctrErrs != nil {
```

Comment:
This probably deserves its own helper, it gets repeated a fair bit

Reply:
fixed